src/commons/api/roas.rs

```diff
@@ -280,22 +286,186 @@ impl Ord for RoaDefinition {
     fn as_ref(&self) -> &TypedPrefix {
         &self.prefix
     }
 }
 
+//------------ RoaConfiguration --------------------------------------------
+
+/// This type defines a saved RoaConfiguration. It includes the actual ROA
+/// payload that needs be authorized in a ROA object as well as other
+/// information that is only visible to users - i.e. an optional comment
+/// field, and in future perhaps other things such as tags used for
+/// classification/monitoring/bpp analysis.
+///
+/// Note that this type is backward compatible with the RoaDefinition type
+/// used until Krill 0.10.0.
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct RoaDefinitions(Vec<RoaDefinition>);
+pub struct RoaConfiguration {
+    #[serde(flatten)]
+    payload: RoaPayload,
+    #[serde(default)] // missing is same as no comment
+    comment: Option<String>,
+}
+
+impl RoaConfiguration {
+    pub fn new(payload: RoaPayload, comment: Option<String>) -> Self {
+        RoaConfiguration { payload, comment }
+    }
+
+    pub fn unpack(self) -> (RoaPayload, Option<String>) {
+        (self.payload, self.comment)
+    }
+
+    pub fn payload(&self) -> RoaPayload {
+        self.payload
+    }
+
+    pub fn comment(&self) -> Option<&String> {
+        self.comment.as_ref()
+    }
+
+    /// Ensures that the payload uses an explicit max length
+    pub fn into_explicit_max_length(self) -> Self {
+        RoaConfiguration {
+            payload: self.payload.into_explicit_max_length(),
+            comment: self.comment,
+        }
+    }
+}
+
+impl fmt::Display for RoaConfiguration {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.payload)?;
+        if let Some(comment) = &self.comment {
+            write!(f, " # {}", comment)?;
+        }
+        Ok(())
+    }
+}
+
+impl FromStr for RoaConfiguration {
+    type Err = AuthorizationFmtError;
+
+    // "192.168.0.0/16 => 64496"
+    // "192.168.0.0/16 => 64496 # my nice ROA"
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut parts = s.splitn(2, '#');
+        let payload_part = parts.next().ok_or_else(|| AuthorizationFmtError::auth(s))?;
+
+        let payload = RoaPayload::from_str(payload_part)?;
+        let comment = parts.next().map(|s| s.trim().to_string());
+
+        Ok(RoaConfiguration { payload, comment })
+    }
+}
+
+impl Ord for RoaConfiguration {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.payload.cmp(&other.payload)
+    }
+}
+
+impl PartialOrd for RoaConfiguration {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl From<RoaPayload> for RoaConfiguration {
+    fn from(payload: RoaPayload) -> Self {
+        RoaConfiguration { payload, comment: None }
+    }
+}
+
+//------------ ConfiguredRoa -----------------------------------------------
+
+/// Defines an existing ROA configuration and related information.
```

Comment:
Above `RoaConfiguration` is documented as "defines a saved RoaConfiguration", how is that different to "Defines an existing ROA configuration"? Perhaps use other wording to make the difference between the purpose of these types clearer.

Reply:
fixed